meson.build

```diff
@@ -0,0 +1,248 @@
+conf.set_quoted('LXCFS_SOURCE_ROOT',	project_source_root)
+conf.set_quoted('BINDIR',		bindir)
+conf.set_quoted('LIBDIR',		libdir)
+conf.set_quoted('LOCALSTATEDIR',	localstatedir)
+conf.set_quoted('LXCFSSHAREDIR',	join_paths(localstatedir, 'lxcfs'))
+conf.set_quoted('LXCCONFDIR',		join_paths(localstatedir, '/lxc/config/common.conf.d'))
+conf.set_quoted('RUNTIME_PATH',		runtimepath)
+conf.set_quoted('SYSCONFDIR',		sysconfdir)
+
+init_script = get_option('init-script')
+
+cc = meson.get_compiler('c')
+meson_build_sh = find_program('tools/meson-build.sh')
+
+want_tests = get_option('tests')
+
+possible_cc_flags = [
+	'-fPIE',
+	'-Wvla',
+	'-std=gnu11',
+	'-fdiagnostics-color',
+	'-Wimplicit-fallthrough=5',
+	'-Wcast-align',
+	'-Wstrict-prototypes',
+	'-fno-strict-aliasing',
+	'-fstack-clash-protection',
+	'-fstack-protector-strong',
+	'--param=ssp-buffer-size=4',
+	'--mcet -fcf-protection',
+	'-Werror=implicit-function-declaration',
+	'-Wlogical-op',
+	'-Wmissing-include-dirs',
+	'-Wold-style-definition',
+	'-Winit-self',
+	'-Wunused-but-set-variable',
+	'-Wno-unused-parameter',
+	'-Wfloat-equal',
+	'-Wsuggest-attribute=noreturn',
+	'-Werror=return-type',
+	'-Werror=incompatible-pointer-types',
+	'-Wformat=2',
+	'-Wshadow',
+	'-Wendif-labels',
+	'-Werror=overflow',
+	'-fdiagnostics-show-option',
+	'-Werror=shift-count-overflow',
+	'-Werror=shift-overflow=2',
+	'-Wdate-time',
+	'-Wnested-externs',
+	'-fasynchronous-unwind-tables',
+	'-fexceptions',
+	'-Warray-bounds',
+	'-Wrestrict',
+	'-Wreturn-local-addr',
+	'-fsanitize=cfi',
+	'-Wstringop-overflow',
+]
+
+possible_link_flags = [
+	'-Wl,--as-needed',
+	'-Wl,--gc-sections',
+	'-Wl,-z,relro',
+	'-Wl,-z,now',
+	'-pie',
+	'-Wl,-fuse-ld=gold',
+]
+
+if meson.version().version_compare('>=0.46')
+	add_project_link_arguments(cc.get_supported_link_arguments(possible_link_flags), language : 'c')
+else
+	add_project_link_arguments(possible_link_flags, language : 'c')
+endif
+
+add_project_arguments(cc.get_supported_arguments(possible_cc_flags), language : 'c')
+
+foreach ident : [
+        ['strlcpy',		'''#include <string.h>'''],
+        ['strlcat',		'''#include <string.h>'''],
+        ['pidfd_send_signal',	'''#include <stdlib.h>
+                                 #include <unistd.h>
+                                 #include <signal.h>
+                                 #include <sys/wait.h>'''],
+        ['pidfd_open',		'''#include <stdlib.h>
+                                 #include <unistd.h>
+                                 #include <signal.h>
+                                 #include <sys/wait.h>'''],
+]
+
+        have = cc.has_function(ident[0], prefix : ident[1], args : '-D_GNU_SOURCE')
+        conf.set10('HAVE_' + ident[0].to_upper(), have)
+endforeach
+
+conf.set('_FILE_OFFSET_BITS', 64)
+
+libfuse = cc.find_library('fuse3', required : false)
+if libfuse.found()
+	conf.set10('HAVE_FUSE3', libfuse.found())
+	conf.set('FUSE_USE_VERSION', 30)
+else
+	libfuse = cc.find_library('fuse')
```

Comment:
Fuse and fuse3 have pkg-config files, fuse.pc and fuse3.pc

I'm not really sure what/how fuse works, but FUSE_USE_VERSION seems like an API level for which the ability to do

```
dependency('fuse', version: '>= 2.6')
```

May be useful?

Reply:
Also https://mesonbuild.com/Wrap-dependency-system-manual.html

You can add a small file at subprojects/fuse3.wrap:

```
[wrap-file]
directory = libfuse-fuse-3.10.5
source_url = https://github.com/libfuse/libfuse/archive/refs/tags/fuse-3.10.5.tar.gz
source_filename = fuse-3.10.5.tar.gz
source_hash = e73f75e58da59a0e333d337c105093c496c0fd7356ef3a5a540f560697c9c4e6

[provide]
fuse3 = libfuse_dep
```

and meson will automatically download and build a copy of fuse3 if it cannot find one on the system.

Reply:
Hm, the problem with this is that `dependency('fuse')` doesn't look for `fuse3` when just specifying `fuse`. No big deal. I'll work around it.